### PR TITLE
HER-23-add-teacher-to-roles-in-user

### PR DESCRIPTION
**Pull request by mhope21**
HER-23: Add `teacher` to roles in user

## Issue

https://raquelanaroman.atlassian.net/browse/HER-23

As a teacher-user, I would like a way to distinguish myself from other types of users in the application.

Acceptance Criteria
* add teacher to User::ROLES

[HER-24](https://github.com/Code-the-Dream-School/herring-team-3/pull/15)

## Changes

User model (app/models/user.rb):
* Add teacher to ROLES

### Review Checklist

- [x] I have documented my code with code comments.

### CODE REVIEW

File: frontend/src/components/auth/Registration.jsx [hunk 1]
```diff
@@ -5,33 +5,45 @@ import { Link, useNavigate } from "react-router-dom";
 export default function Registration() {
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
+  // Set useState for isTeacher with a default value of false
```
Comment (raquii) [inline frontend/src/components/auth/Registration.jsx:8]: when leaving code comments, comment things that are not obvious at first glance. This doesn't need to be commented.

```suggestion
```

File: frontend/src/components/auth/Registration.jsx [hunk 2]
```diff
@@ -5,33 +5,45 @@ import { Link, useNavigate } from "react-router-dom";
...
+  const handleClick = () => {
+    setIsTeacher((prevIsTeacher) => {
+      const newIsTeacher = !prevIsTeacher;
+      if (newIsTeacher) {
+        console.log("I am a Teacher");
+      } else {
+        console.log("I am not a teacher");
+      }
+      return newIsTeacher;
+    });
```
Comment (raquii) [inline frontend/src/components/auth/Registration.jsx:28]: see comment on #15

**Review by raquii — DISMISSED** [review]
comments are recommendations. lgtm

**Comment by mhope21** [comment]
Merge conflicts resolved. Fixed missing closing div on Registration page.

**Review by dewi-anggraini — APPROVED** [review]
lgtm

**Review by raquii — APPROVED** [review]

**Pull request merged**